components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java

```diff
@@ -1429,10 +1428,16 @@
                     "FROM ENDPOINT_REG_ENTRY AS E, ENDPOINT_REG AS R " +
                     "WHERE E.REG_ID=R.ID AND R.UUID=?";
 
+    public static final String ADD_ENDPOINT_REGISTRY_ENTRY_SQL =
+            "INSERT INTO ENDPOINT_REG_ENTRY (UUID, ENTRY_NAME, SERVICE_URL, DEFINITION_TYPE, DEFINITION_URL, METADATA," +
+                    "SERVICE_TYPE, ENDPOINT_DEFINITION, REG_ID) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)";
 
     public static final String IS_ENDPOINT_REGISTRY_NAME_EXISTS = "SELECT COUNT(UUID) AS ENDPOINT_REGISTRY_COUNT" +
             " FROM ENDPOINT_REG WHERE LOWER(REG_NAME) = LOWER(?) AND TENANT_ID = ?";
 
+    public static final String IS_ENDPOINT_REGISTRY_ENTRY_NAME_EXISTS = "SELECT COUNT(UUID) AS REGISTRY_ENTRY_COUNT" +
```

Comment:
is there an issue in doing a getEndpointRegistryByUUID instead of introducing a new SQL query to check the existance?

Reply:
This query is to check whether the given entry name already exists.

Reply:
**getEndpointRegistryByUUID** method cannot be used for this purpose I think. Here, the existence of the entry is checked using its' **name** not using **UUID**. Validating the unique constraint of (ENTRY_NAME, REG_ID).

Reply:
got it.